sdk/core/core-amqp/src/connectionConfig/connectionConfig.ts

```diff
@@ -91,7 +91,7 @@ export namespace ConnectionConfig {
     }
 
     if (!parsedCS.Endpoint.endsWith("/")) parsedCS.Endpoint += "/";
-
+    parsedCS.Endpoint = parsedCS.Endpoint.replace(new RegExp(".*://"), "sb://");
```

Comment:
Can we tighten the regex? Maybe something like:
```suggestion
    parsedCS.Endpoint = parsedCS.Endpoint.replace(new RegExp("^\w+://"), "sb://");
```
(Ensures that the endpoint starts with at least 1 alphanumeric characters or underscore)

Also curious why is this change needed? I've seen examples of connecting to event hubs using rhea with the `amqps` scheme isntead of `sb` but does that break at this layer?

Reply:
`sb://` is not used internally when hitting the service to my understanding.

We have a user-issue #7907 - auth failed due to the SDK's expectation that the endpoint in the connection-string should start with `sb://` whereas the user-provided one has `https://`.
This is to fix the above issue.

.NET handles this by replacing the scheme to `sb://` in the SDK, this PR attempts to align with the same.